### PR TITLE
[PD-3900] Add a debounce for TJD

### PR DESCRIPTION
**Pull request by vigowei**
## What's this PR do?
Adds a debounce to run_trial_report, so it won't be triggered every user edit 

### Front-End

- [ ] Works on IE 11 and Edge
- [ ] Works on Mobile
- [ ] Unit tests added or ticket exists to add tests

### Separation of Concerns
n/a

### Tests
n/a

### Migrations
n/a

### CODE REVIEW

**Comment by vigowei** [comment]
closing to fix tests

File: gulp/src/reporting/components/SetUpReport.jsx [hunk 1]
```diff
@@ -111,10 +114,10 @@ class SetUpReport extends Component {
     this.dispatchUpdateFilterWithDependencies();
   }
 
-  dispatchUpdateFilterWithDependencies() {
+  dispatchUpdateFilterWithDependencies = debounce(() => {
     const {dispatch, filterInterface, reportDataId} = this.props;
     return dispatch(doUpdateFilterWithDependencies(filterInterface, reportDataId));
-  }
+  }, 500);
```
Comment (aljordan82) [inline gulp/src/reporting/components/SetUpReport.jsx]: I've tested and it works.
There is a `typingDebounce` function that we use, but it's hardcoded to 300ms. We can you that if you are fine with 300ms.

File: gulp/src/reporting/components/SetUpReport.jsx [hunk 2]
```diff
@@ -402,7 +412,7 @@ class SetUpReport extends Component {
                 <TextField
                   name={currentFilter[col.filter] || ''}
                   value={currentFilter[col.filter] || ''}
-                  onChange={v => this.dispatchFilterAction(setSimpleFilterAction(col.filter, v.target.value))}
+                  onChange={v => this.dispatchFilterAction(setSimpleFilterAction(col.filter, v.target.value), 'debounced')}
```
Comment (aljordan82) [inline gulp/src/reporting/components/SetUpReport.jsx]: I would use true and you can add a comment like .. , true /* debounce */)

Comment (vigowei) [reply gulp/src/reporting/components/SetUpReport.jsx]: @aljordan82 either way would be readable right? if we allow anything truthy in the function, how does this make a difference?

Comment (aljordan82) [reply gulp/src/reporting/components/SetUpReport.jsx]: True, there is not difference, it is just not expected to compare against a string.

**Comment by koonsies** [comment]
I'm getting some weird results with the debounce. None of the fields are causing a trial report to be run instantly. When I change a multi-select box, a trial report isn't fired until after a looonngg delay. Debounce seems to work for the text field, but it's also way too long.

**Pull request merged**